### PR TITLE
feat: add `sequential_transactions_by_sender` method for `MockTransactionSet`

### PR DESCRIPTION
**Pull request by tcoratger**
Added a new method `sequential_transactions_by_sender` to the `MockTransactionSet` struct. 

This method generates a sequence of transactions for a specified sender, creating a chain based on the provided transaction count.

### CODE REVIEW

File: crates/transaction-pool/src/test_utils/mock.rs [hunk 1]
```diff
@@ -1156,7 +1160,27 @@ impl MockTransactionSet {
...
+    ) -> Self {
+        let mut txs = Vec::with_capacity(tx_count);
+        let mut current_tx = MockTransaction::new_from_type(tx_type).with_sender(sender);
+
+        for _ in 0..tx_count {
+            txs.push(current_tx.clone());
+            current_tx = current_tx.next();
+        }
+
+        Self::new(txs)
```
Comment (Rjected) [inline crates/transaction-pool/src/test_utils/mock.rs:1183]: this could probably just call `dependent` with a `from_nonce` of 0

**Review by Rjected — CHANGES_REQUESTED** [review]
I like this overall, just one comment so we have less code duplication

File: crates/transaction-pool/src/test_utils/mock.rs (hunk 1 above)
Comment (tcoratger) [reply crates/transaction-pool/src/test_utils/mock.rs]: yes, just done

**Review by Rjected — APPROVED** [review]
LGTM, thanks!

**Review by mattsse — APPROVED** [review]
ty!

**Pull request merged**